practice/src/main/java/practice/tasks/multithreading/MultithreadingTask.java

```diff
@@ -1,11 +1,42 @@
-package practice.tasks;
+package practice.tasks.multithreading;
 
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
 
 public class MultithreadingTask {
 
+    public int elementsSumWithThreadExecutor(int[] numbers, int threads) throws ExecutionException, InterruptedException {
+        int resultElementsSum = 0;
+        ExecutorService executor = Executors.newFixedThreadPool(threads);
+
+        List<int[]> subArrays = divideArray(numbers, threads);
+        List<Future<Integer>> futures = new ArrayList<>();
+
+        for (int[] subArray : subArrays) {
+            Callable<Integer> taskToGetSumsOfSubArrays = () -> {
+                int eachArraySum = 0;
+                for (int number : subArray) {
+                    eachArraySum += number;
+                }
+                return eachArraySum;
+            };
+            futures.add(executor.submit(taskToGetSumsOfSubArrays));
```

Comment:
The idea of thread pools is to avoid of creation of new threads in every computation. In your code you still create new threads in every `elementsSumWithThreadExecutor` execution.
Check this case:
1.  You execute this method in your service every 10 seconds and ask to use 5 threads for computation.
2. In this case, in your implementation every 10 seconds 5 new threads are created and destroyed after the method is completed.

What we need here, is to reuse the same 5 threads every time.